roofit/roofitcore/src/RooConvGenContext.cxx

```diff
@@ -142,17 +138,24 @@ RooConvGenContext::RooConvGenContext(const RooNumConvPdf &model, const RooArgSet
          << " for generation of observable(s) " << vars << endl ;
 
   // Create generator for physics X truth model
-  _pdfVarsOwned = (RooArgSet*) model.conv().clonePdf().getObservables(&vars)->snapshot(true) ;
-  _pdfVars = new RooArgSet(*_pdfVarsOwned) ;
-  _pdfGen = ((RooAbsPdf&)model.conv().clonePdf()).genContext(*_pdfVars,prototype,auxProto,verbose) ;
-  _pdfCloneSet = 0 ;
+  {
+    RooArgSet clonedPdfObservables;
+    model.conv().clonePdf().getObservables(&vars, clonedPdfObservables);
```

Comment:
Just for informing the reviewer: there was a memory leak here because `getObservables()` returns an owning pointer as it was used before. Now, the other overload that uses an output parameter is used to avoid heap allocations.